### PR TITLE
Fix: typos and formatting issues on documentation

### PR DESCRIPTION
**Pull request by RuggeroCino**
## Description

As I was reading the [OSx documentation](https://devs.aragon.org/docs/osx/), I found some small typos and formatting issues that have been fixed on this PR.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have updated the `DEPLOYMENT_CHECKLIST` file in the root folder.
- [ ] I have updated the `UPDATE_CHECKLIST` file in the root folder.

### CODE REVIEW

File: packages/contracts/docs/osx/01-how-it-works/01-core/03-plugins/index.md [hunk 1]
```diff
@@ -29,15 +29,13 @@ Lern more about the different [plugin types](../../../02-how-to-guides/02-plugin
 
 This raises questions on how the DAO manages plugins and who actually owns plugins.
 
-### How does the DAO Manage a Plugin?
+### How does the DAO manage a Plugin?
```
Comment (Michael-A-Heuer) [inline packages/contracts/docs/osx/01-how-it-works/01-core/03-plugins/index.md:32]: Correct according to https://capitalizemytitle.com/

Comment (Michael-A-Heuer) [inline packages/contracts/docs/osx/01-how-it-works/01-core/03-plugins/index.md:32]:
```suggestion
### How does the DAO Manage a Plugin?
```

**Review by Michael-A-Heuer — APPROVED** [review]

File: packages/contracts/docs/osx/01-how-it-works/01-core/03-plugins/index.md [hunk 2]
```diff
@@ -29,15 +29,13 @@ Lern more about the different [plugin types](../../../02-how-to-guides/02-plugin
 
 This raises questions on how the DAO manages plugins and who actually owns plugins.
 
-### How does the DAO Manage a Plugin?
+### How Does the DAO Manage a Plugin?
```
Comment (brickpop) [inline packages/contracts/docs/osx/01-how-it-works/01-core/03-plugins/index.md:32]: Where is the issue with "does"?

File: packages/contracts/docs/osx/01-how-it-works/01-core/index.md [hunk 3]
```diff
@@ -26,8 +26,6 @@ In a nutshell, your Aragon OSx DAO consists of three pieces:
 
 The underlying smart contracts constitute **the core contracts** of the Aragon OSx DAO framework.
 
-The following graphic shows how an exemplary DAO setup, where the
```
Comment (brickpop) [inline packages/contracts/docs/osx/01-how-it-works/01-core/index.md:29]: @Michael-A-Heuer shouldn't we rather complete the sentence?

File: packages/contracts/docs/osx/01-how-it-works/01-core/03-plugins/index.md (hunk 2 above)
Comment (RuggeroCino) [reply packages/contracts/docs/osx/01-how-it-works/01-core/03-plugins/index.md:32]: As noted by Michael on [this discussion](https://github.com/aragon/osx/pull/434#discussion_r1266716602), headings should be in title case

File: packages/contracts/docs/osx/01-how-it-works/01-core/index.md (hunk 3 above)
Comment (RuggeroCino) [reply packages/contracts/docs/osx/01-how-it-works/01-core/index.md:29]: The image already has a caption below, I found this paragraph redundant:
Screenshot 2023-07-20 at 08 44 44

**Review by Michael-A-Heuer — APPROVED** [review]

**Pull request merged**